File: 0401_binary_watch/solution.py
# LeetCode 0401 - Binary Watch
# https://leetcode.com/problems/binary-watch/

from typing import List


class Solution:
    def readBinaryWatch(self, turnedOn: int) -> List[str]:
        result: list[str] = []
        for hour in range(12):
            for minute in range(60):
                if bin(hour).count("1") + bin(minute).count("1") == turnedOn:
                    result.append(f"{hour}:{minute:02d}")
        return result

File: 0401_binary_watch/solution.rb
# LeetCode 0401 - Binary Watch
# https://leetcode.com/problems/binary-watch/

class Solution
  def read_binary_watch(turned_on)
    result = []
    (0...12).each do |hour|
      (0...60).each do |minute|
        if hour.to_s(2).count("1") + minute.to_s(2).count("1") == turned_on
          result << format("%d:%02d", hour, minute)
        end
      end
    end
    result
  end

  alias_method :readBinaryWatch, :read_binary_watch
end
